client/src/main/scala/caliban/client/__Value.scala

```diff
@@ -20,6 +24,9 @@ sealed trait __Value { self =>
 }
 
 object __Value {
+  private final val MaxDepth                                    = 512
```

Comment:
Do we think there might be cases that the users want to grab more than 512 fields deep? In the server we were comfortable in having this since it's up to the user to choose a JSON implementation, but here we're "forcing" one on them - so I'm not sure whether we need to think of any potential edge cases

Reply:
I'm not sure, that seems like it would be _really_ deep as a structure. Maybe we could provide it as an `implicit` would that be better? 

```scala
case class MaxDepth(depth: Int)

object MaxDepth {
  implicit val default: MaxDepth = MaxDepth(512)
}
```

Reply:
To be honest I agree with you, these are _really_ nested objects. I doubt there are many, if any, real use-cases where one would need to query this deep. 

I think we could probably get away with increasing this value to 1023 and see if anyone ever requests for this limit to be increased or made configurable?